src/NumericalAlgorithms/Interpolation/MultiLinearSpanInterpolation.hpp

```diff
@@ -0,0 +1,397 @@
+    allow_extrapolation_above_data[whichDimension] = value;
+  }
+
+  void extrapolateBelowData(size_t whichDimension, bool value) {
+    allow_extrapolation_below_data[whichDimension] = value;
+  }
+
+  /// Determine relative index bracketing for non-uniform spacing
+  size_t findIndexGeneral(const size_t whichDimension,
+                          const double& targetPoints) const;
+
+  //! Determine relative index bracketing for non-uniform spacing
+  size_t findIndexUniform(const size_t whichDimension,
+                          const double& targetPoints) const;
+
+  size_t findIndex(const size_t whichDimension,
+                   const double& targetPoints) const {
+    if constexpr (UniformSpacing) {
+      return findIndexUniform(whichDimension, targetPoints);
+    } else {
+      return findIndexGeneral(whichDimension, targetPoints);
+    }
+  };
+
+  /// Compute interpolation weights for 1D tables
+  weight_t<1> getWeights(const double x1) const;
+
+  /// Compute interpolation weights for 2D tables
+  weight_t<2> getWeights(const double x1, const double x2) const;
+
+  /// Compute interpolation weights for 3D tables
+  weight_t<3> getWeights(const double x1, const double x2,
+                         const double x3) const;
+
+  double interpolate(weight_t<Dimension> const& weights,
+                     size_t whichVariable = 0) const {
+    double result = 0;
+
+    for (size_t nn = 0; nn < weights.weights.size(); ++nn)
+      result += weights.weights[nn] *
+                y[whichVariable + NumberOfVariables * weights.index[nn]];
+
+    return result;
+  };
+
+  template <size_t... VariablesToInterpolate>
+  std::array<double, sizeof...(VariablesToInterpolate)> interpolate(
+      weight_t<Dimension> const& weights) const {
+    static_assert(sizeof...(VariablesToInterpolate) <= NumberOfVariables,
+                  "You are trying to interpolate more variables than this "
+                  "container holds.");
+
+    return std::array<double, sizeof...(VariablesToInterpolate)>{
+        interpolate(weights, VariablesToInterpolate)...};
+  }
+
+  template <size_t NumberOfVariablesToInterpolate, typename... T,
+            Requires<(std::is_floating_point_v<typename std::remove_cv_t<T>> and
+                      ...)> = nullptr>
+  std::array<double, NumberOfVariablesToInterpolate> interpolate(
+      Index<NumberOfVariablesToInterpolate>& VariablesToInterpolate,
+      T const&... targetPoints) const {
+    static_assert(
+        sizeof...(T) == Dimension,
+        "You need to provide the correct number of interpolation points");
+    static_assert(NumberOfVariablesToInterpolate <= NumberOfVariables,
+                  "You are trying to interpolate more variables than this "
+                  "container holds.");
+
+    auto weights = getWeights(targetPoints...);
+
+    std::array<double, NumberOfVariablesToInterpolate> interpolatedValues;
+
+    for (size_t nn = 0; nn < NumberOfVariablesToInterpolate; ++nn)
```

Comment:
we prefer braces around all loops, even single line ones. We've had bugs in SpEC where people indented "into" the for loop or if statement. Even if the code is technically what was meant, it's just harder for that to be clear